tests/conftest.py

```diff
@@ -31,6 +31,7 @@ def pytest_configure(config):
 def pytest_addoption(parser):
     parser.addoption("--torch_ver", default=None, type=str)
     parser.addoption("--cuda_ver", default=None, type=str)
+    parser.addoption("--enable-compile-mode", action="store_true", help="Run both compiled/non-compiled versions")
```

Comment:
Perhaps this would be less ambiguous to what it implies?

```suggestion
    parser.addoption("--enable-compile-mode", action="store_true", help="Run both compiled/non-compiled versions")
        parser.addoption("--enable-torch-compile", action="store_true", help="Run both compiled/non-compiled versions")
```

Reply:
@stas00 regarding your questions:
 1. There are some basic torch.compile tests which are always run (e.g. deepspeed-fork/tests/unit/runtime/compile/util.py ). this PR is a way to add more torch.compile coverage for a lot more scenarios using the same logic as the original tests.
 2. I'm sure there are ways to do this other than passing a parameter, but this is standard way to parametrize pytorch tests.  it is printed clearly in the pytest logs which config is running in this way. choosing another approach may have some advantages, but could make it less clear which scenario is actually running (e.g. in your example this wouldn't explicitly be printed in the test name).
 example:
 7.84s call     unit/checkpoint/test_latest_checkpoint.py::TestLatestCheckpoint::test_existing_latest[**False**]

I'm not opposed to a different approach, but I'm not convinced it offers an actual advantage. will wait for feedback from @tohtana and @tjruwase. 
 
regarding the flag, I used the one suggested by @tohtana and it makes some semantic sense since this flag enables compile mode in the tests it doesn't enable torch.compile in general ( which might make it more confusing). I don't mind changing this to whatever there is a consensus on.